src/test/harness.js

```diff
@@ -900,9 +901,13 @@ const commands = mapValues(testCommands, (command, name) => {
 
 async function describe(description, cbk) {
   console.log(`# Test ${description}`);
-  await start();
-  await cbk();
-  finish();
+  try {
+    await start();
+    await cbk();
+    finish();
+  } catch (e) {
+    window.TestFinished = e.message;
```

Comment:
What is this for?  This needs a comment at least.

Reply:
Definitely. This is a _clever_ way to pass in the reason why the test failed, but I think there's a simpler way to do it.